pkg/asset/installconfig/azure/validation.go

```diff
@@ -711,18 +711,15 @@ func validateResourceGroup(client API, fieldPath *field.Path, platform *aztypes.
 		allErrs = append(allErrs, field.Invalid(fieldPath.Child("resourceGroupName"), platform.ResourceGroupName, fmt.Sprintf("resource group has conflicting tags %s", strings.Join(conflictingTagKeys, ", "))))
 	}
 
-	// ARO provisions Azure resources before resolving the asset graph.
-	if !platform.IsARO() {
-		ids, err := client.ListResourceIDsByGroup(context.TODO(), platform.ResourceGroupName)
-		if err != nil {
-			return append(allErrs, field.InternalError(fieldPath.Child("resourceGroupName"), fmt.Errorf("failed to list resources in the resource group: %w", err)))
-		}
-		if l := len(ids); l > 0 {
-			if len(ids) > 2 {
-				ids = ids[:2]
-			}
-			allErrs = append(allErrs, field.Invalid(fieldPath.Child("resourceGroupName"), platform.ResourceGroupName, fmt.Sprintf("resource group must be empty but it has %d resources like %s ...", l, strings.Join(ids, ", "))))
+	ids, err := client.ListResourceIDsByGroup(context.TODO(), platform.ResourceGroupName)
```

Comment:
I don't think we can just enable this code unconditionally. Rather, we need to move it to somewhere that the ARO installer naturally doesn't run instead of having it behind an explicit flag.

A possible pattern might be to move this validation to the PlatformProvisionCheck asset, which doesn't run until the `create cluster` phase (which ARO doesn't use). @patrickdillon do you have any thoughts on that?

Reply:
In general, validations should go here if they apply to both UPI & IPI and in PlatformProvisionCheck if IPI-only. I often find it very difficult to reason about this distinction...

Moving to PlatformProvisionCheck makes sense, but talking through it:

The risk we have by moving the validation is that somebody creates a UPI cluster in a resource group that contains other resources and then runs `destroy cluster` and wipes out the whole resource group.

Initially my thought was "nobody doing UPI is going to do that" but the result is pretty catastrophic, so that makes me hesitant...

Doesn't ARO set a special marker on the resource group (managed by or something?) Could we inspect the resource group and see if it contains that special ARO flag, and skip the validation just for them? LMK if that idea isn't clear.

Reply:
Following up on my previous comment,  ARO does indeed set [ManagedBy](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.management.resourcemanager.models.resourcegroup.managedby?view=azure-dotnet-legacy) on the rg, but I believe the value is non-determinsitic and non-discoverable. 

Does ARO use the installer for destroy? If not, I would consider skipping this check if ManagedBy is set and disallowing destroy if it is set. 

If not,